client/src/pages/SignUp.js

```diff
@@ -0,0 +1,245 @@
+import React, { Component } from "react";
+import {
+  Button,
+  Typography,
+  TextField,
+  Card,
+  CardContent,
+  CardActions,
+  Grid,
+} from "@material-ui/core";
+import { Link } from "react-router-dom";
+import { withStyles } from "@material-ui/core/styles";
+import BackgroundImg from "../Assets/images/signUpBkg.png";
+
+const regPattern = new RegExp(
+  /^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i
+);
+
+const signUpPageStyle = (theme) => ({
+  signUpContainer: {
+    fontFamily: '"Montserrat"',
+    margin: theme.spacing(2),
+    height: "100vh",
+  },
+  card: {
+    width: "50%",
+    height: "75%",
+    marginLeft: theme.spacing(20),
+    textAlign: "center",
+  },
+  title: {
+    fontSize: 25,
+    fontWeight: "bold",
+    textAlign: "center",
+  },
+  button: {
+    color: "white",
+    width: theme.spacing(15),
+    background: "#FF510C", //Brighter Orange
+    "&:hover": {
+      backgroundColor: "#FF510C", //Brighter Orange
+    },
+  },
+  titleLogo: {
+    width: "50%",
+    marginLeft: theme.spacing(20),
+    paddingBottom: theme.spacing(2),
+    textAlign: "center",
+  },
+  leftSideContainer: {
+    width: "50%",
+    height: "100vh",
+    float: "left",
+  },
+  rightSideContainer: {
+    width: "50%",
+    height: "100vh",
+    [theme.breakpoints.up("md")]: {
+      backgroundImage: `url(${BackgroundImg})`,
+      backgroundSize: "cover",
+      backgroundPosition: "left",
+    },
+    float: "left",
+  },
+  textField: {
+    margin: theme.spacing(2),
+    width: "30ch",
+  },
+  whiteText: {
+    color: "white",
+  },
+  marginLeft_2: {
+    margin: "2em",
+  },
+  linkNoDecoration: {
+    textDecoration: "none",
+  },
+});
+
+class SignUpPage extends Component {
```

Comment:
This needs to be made into a functional component

Reply:
Done.